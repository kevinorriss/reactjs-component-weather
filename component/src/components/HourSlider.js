import React from 'react'
import PropTypes from 'prop-types'
import Slider from 'rc-slider'
import axios from 'axios'
import moment from 'moment-timezone'
import 'rc-slider/assets/index.css'

class HourSlider extends React.Component {
    constructor(props) {
        super(props)

        // bind this to the functions
        this.getForecast = this.getForecast.bind(this)

        // set the initial state
        this.state = {
            markCount: 12,
            markStep: 2,
            hourStep: 2,
            marks: {},
            disabled: true
        }
    }

    componentDidUpdate(prevProps) {
        // if the lat/long was null and has been updated, we have the location
        if (!prevProps.latitude && !prevProps.longitude && this.props.latitude && this.props.longitude) {
            this.getForecast(this.props.latitude, this.props.longitude)
        }
    }

    async requestForecast(url) {
        return await axios.get(url)
    }

    async getForecast(latitude, longitude) {
        let response
        try {
            // call the REST API to get the forecast data for the provided location
            const url = `${this.props.forecastURL}?latitude=${latitude}&longitude=${longitude}`
            response = await this.requestForecast(url)
        } catch (error) {
            // fire the error callback function and return
            this.props.onForecastError('Error getting forecast response')
            return
        }

        // destructure the forecast
        const { timezone, hourly } = response.data

        const maxMark = (this.state.markCount - 1) * this.state.hourStep

        // check that we have enough hourly forecasts for each step
        if (hourly.length < maxMark) {
            this.props.onForecastError("Hourly forecast not covered")
            return
        }

        // fire the forecast callback function
        this.props.onForecastReceived(response.data)

        let marks = {}
        let count = 0
        for (let i = 0; i <= maxMark; i += this.state.hourStep) {
            if (count % this.state.markStep == 0) {
                marks[count] = moment.unix(hourly[i].time).tz(timezone).format('HH:mm')
            } else {
                marks[count] = ''
            }
            count++;
        }

        this.setState((prevState) => ({
            ...prevState,
            marks,
            disabled: false
        }))
    }

    render() {
        return (
            <Slider marks={this.state.marks}
                step={1}
                max={this.state.markCount-1}
                included={false}
                disabled={this.state.disabled}
                onChange={(index) => { this.props.onSliderChange(index*this.state.hourStep) }}
            />
        )
    }
}

HourSlider.propTypes = {
    latitude: PropTypes.number,
    longitude: PropTypes.number,
    onForecastReceived: PropTypes.func.isRequired,
    onForecastError: PropTypes.func.isRequired,
    onSliderChange: PropTypes.func.isRequired,
    forecastURL: PropTypes.string.isRequired,
}

export default HourSlider
