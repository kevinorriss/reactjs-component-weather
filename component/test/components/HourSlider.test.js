import axios from 'axios'
import HourSlider from '../../src/components/HourSlider'

const getForecast = jest.spyOn(HourSlider.prototype, 'getForecast')

// setup variables before each test
let positionOne
let forecast
beforeEach(() => {
    // test location
    positionOne = {
        latitude: 51.505455,
        longitude: -0.075356
    }

    forecast = {
        timezone: 'Europe/London',
        hourly: [
            {time: 1582848000, summary: 'Rain', icon: 'rain', precipProbability: 0.85, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582851600, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582855200, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582858800, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582862400, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582866000, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582869600, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582873200, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582876800, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582880400, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582884000, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582887600, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582891200, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582894800, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582898400, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582902000, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582905600, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582909200, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582912800, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582916400, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582920000, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
            {time: 1582923600, summary: 'Rain', icon: 'rain', precipProbability: 0.82, temperature: 7.34, humidity: 0.82, windSpeed: 6.31},
        ],
        daily: [
            {time: 1582416000, summary: 'Light rain', icon: 'rain', temperatureMin: 6.37, temperatureMax: 13.47},
            {time: 1582516000, summary: 'Overcast', icon: 'cloudy', temperatureMin: 4.23, temperatureMax: 12.42}
        ]
    }

    // mock the external API response
    axios.get = jest.fn()
    axios.get.mockImplementationOnce(() => Promise.resolve({ data: forecast}))
})

afterEach(() => {
    getForecast.mockClear()
})

it('Should render initial state', () => {
    // create the callback functions
    const onForecastReceived = jest.fn()
    const onForecastError = jest.fn()
    const onSliderChange = jest.fn()

    // shallow render the component
    const wrapper = mount(
        <HourSlider
            onForecastReceived={onForecastReceived}
            onForecastError={onForecastError}
            onSliderChange={onSliderChange}
            forecastURL=""/>
    )

    // component should still be disabled
    expect(wrapper.state('disabled')).toEqual(true)
    expect(wrapper.state('marks')).toEqual({})

    // should render as expected
    expect(wrapper).toMatchSnapshot()
})

it('Should call error callback on external API request error', (done) => {
    axios.get = jest.fn()
    axios.get.mockImplementationOnce(() => Promise.reject())

    // create the callback functions
    const onForecastReceived = jest.fn()
    const onForecastError = jest.fn()
    const onSliderChange = jest.fn()

    // shallow render the component
    const wrapper = mount(
        <HourSlider
            onForecastReceived={onForecastReceived}
            onForecastError={onForecastError}
            onSliderChange={onSliderChange}
            forecastURL=""/>
    )

    // set the lat/long props
    wrapper.setProps({ latitude: positionOne.latitude, longitude: positionOne.longitude })

    // set a short delay before testing callbacks
    setTimeout(() => {
        // callbacks shouldn't have been called
        expect(onForecastReceived).toHaveBeenCalledTimes(0)
        expect(onSliderChange).toHaveBeenCalledTimes(0)

        // error callback should have been called
        expect(onForecastError).toHaveBeenCalledTimes(1)
        expect(onForecastError).toHaveBeenCalledWith('Error getting forecast response')

        // expect the initial state
        expect(wrapper.state('disabled')).toEqual(true)
        expect(wrapper.state('marks')).toEqual({})

        // should render correctly
        expect(wrapper).toMatchSnapshot()

        // tell jest the test is complete
        done()
    }, 20)
})

it('should handle when response contains less hourly data than expected', (done) => {
    // create the callback functions
    const onForecastReceived = jest.fn()
    const onForecastError = jest.fn()
    const onSliderChange = jest.fn()

    // set the hourly array to contain only the first two elements
    forecast.hourly.splice(2)

    // mock the external API call with the partial hourly forecast
    axios.get = jest.fn()
    axios.get.mockImplementationOnce(() => Promise.resolve({ data: forecast}))

    // shallow render the component
    const wrapper = mount(
        <HourSlider
            onForecastReceived={onForecastReceived}
            onForecastError={onForecastError}
            onSliderChange={onSliderChange}
            forecastURL=""/>
    )

    // set the lat/long props
    wrapper.setProps({ latitude: positionOne.latitude, longitude: positionOne.longitude })

    // set a short delay before testing callbacks
    setTimeout(() => {
        // callbacks shouldn't have been called
        expect(onForecastReceived).toHaveBeenCalledTimes(0)
        expect(onSliderChange).toHaveBeenCalledTimes(0)

        // error callback should have been called
        expect(onForecastError).toHaveBeenCalledTimes(1)
        expect(onForecastError).toHaveBeenCalledWith("Hourly forecast not covered")

        // expect the initial state
        expect(wrapper.state('disabled')).toEqual(true)
        expect(wrapper.state('marks')).toEqual({})

        // should render correctly
        expect(wrapper).toMatchSnapshot()

        // tell jest the test is complete
        done()
    }, 20)
})

it('should handle valid forecast response', (done) => {
    // create the callback functions
    const onForecastReceived = jest.fn()
    const onForecastError = jest.fn()
    const onSliderChange = jest.fn()

    // shallow render the component
    const wrapper = mount(
        <HourSlider
            onForecastReceived={onForecastReceived}
            onForecastError={onForecastError}
            onSliderChange={onSliderChange}
            forecastURL=""/>
    )

    // set the lat/long props
    wrapper.setProps({ latitude: positionOne.latitude, longitude: positionOne.longitude })

    // set a short delay before testing callbacks
    setTimeout(() => {
        // success callback should have been called
        expect(onForecastReceived).toHaveBeenCalledTimes(1)
        expect(onForecastReceived).toHaveBeenCalledWith(forecast)

        // component should no longer be disabled
        expect(wrapper.state('disabled')).toEqual(false)

        // expect the marks to match the test data
        expect(wrapper.state('marks')).toEqual({
            '0': '00:00',
            '1': '',
            '2': '04:00',
            '3': '',
            '4': '08:00',
            '5': '',
            '6': '12:00',
            '7': '',
            '8': '16:00',
            '9': '',
            '10': '20:00',
            '11': ''
        })

        // should render correctly
        expect(wrapper).toMatchSnapshot()
        
        // tell jest the test is complete
        done()
    }, 20)
})