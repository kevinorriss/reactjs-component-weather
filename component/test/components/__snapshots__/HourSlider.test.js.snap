// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should call error callback on external API request error 1`] = `
<HourSlider
  forecastURL=""
  latitude={51.505455}
  longitude={-0.075356}
  onForecastError={
    [MockFunction] {
      "calls": Array [
        Array [
          "Error getting forecast response",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onForecastReceived={[MockFunction]}
  onSliderChange={[MockFunction]}
>
  <ComponentEnhancer(undefined)
    activeDotStyle={Object {}}
    className=""
    disabled={true}
    dotStyle={Object {}}
    dots={false}
    handle={[Function]}
    handleStyle={
      Array [
        Object {},
      ]
    }
    included={false}
    marks={Object {}}
    max={11}
    min={0}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    railStyle={Object {}}
    reverse={false}
    step={1}
    trackStyle={
      Array [
        Object {},
      ]
    }
    vertical={false}
  >
    <div
      className="rc-slider rc-slider-disabled"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <Track
        className="rc-slider-track"
        included={false}
        length={0}
        offset={0}
        reverse={false}
        style={Object {}}
        vertical={false}
      />
      <Steps
        activeDotStyle={Object {}}
        dotStyle={Object {}}
        dots={false}
        included={false}
        lowerBound={0}
        marks={Object {}}
        max={11}
        min={0}
        prefixCls="rc-slider"
        reverse={false}
        step={1}
        upperBound={0}
        vertical={false}
      >
        <div
          className="rc-slider-step"
        />
      </Steps>
      <Handle
        className="rc-slider-handle"
        disabled={true}
        key="0"
        max={11}
        min={0}
        offset={0}
        prefixCls="rc-slider"
        reverse={false}
        style={Object {}}
        value={0}
        vertical={false}
      >
        <div
          aria-disabled={true}
          aria-valuemax={11}
          aria-valuemin={0}
          aria-valuenow={0}
          className="rc-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="slider"
          style={
            Object {
              "left": "0%",
              "right": "auto",
              "transform": "translateX(-50%)",
            }
          }
          tabIndex={null}
        />
      </Handle>
      <Marks
        className="rc-slider-mark"
        included={false}
        lowerBound={0}
        marks={Object {}}
        max={11}
        min={0}
        onClickLabel={[Function]}
        reverse={false}
        upperBound={0}
        vertical={false}
      >
        <div
          className="rc-slider-mark"
        />
      </Marks>
    </div>
  </ComponentEnhancer(undefined)>
</HourSlider>
`;

exports[`Should render initial state 1`] = `
<HourSlider
  forecastURL=""
  onForecastError={[MockFunction]}
  onForecastReceived={[MockFunction]}
  onSliderChange={[MockFunction]}
>
  <ComponentEnhancer(undefined)
    activeDotStyle={Object {}}
    className=""
    disabled={true}
    dotStyle={Object {}}
    dots={false}
    handle={[Function]}
    handleStyle={
      Array [
        Object {},
      ]
    }
    included={false}
    marks={Object {}}
    max={11}
    min={0}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    railStyle={Object {}}
    reverse={false}
    step={1}
    trackStyle={
      Array [
        Object {},
      ]
    }
    vertical={false}
  >
    <div
      className="rc-slider rc-slider-disabled"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <Track
        className="rc-slider-track"
        included={false}
        length={0}
        offset={0}
        reverse={false}
        style={Object {}}
        vertical={false}
      />
      <Steps
        activeDotStyle={Object {}}
        dotStyle={Object {}}
        dots={false}
        included={false}
        lowerBound={0}
        marks={Object {}}
        max={11}
        min={0}
        prefixCls="rc-slider"
        reverse={false}
        step={1}
        upperBound={0}
        vertical={false}
      >
        <div
          className="rc-slider-step"
        />
      </Steps>
      <Handle
        className="rc-slider-handle"
        disabled={true}
        key="0"
        max={11}
        min={0}
        offset={0}
        prefixCls="rc-slider"
        reverse={false}
        style={Object {}}
        value={0}
        vertical={false}
      >
        <div
          aria-disabled={true}
          aria-valuemax={11}
          aria-valuemin={0}
          aria-valuenow={0}
          className="rc-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="slider"
          style={
            Object {
              "left": "0%",
              "right": "auto",
              "transform": "translateX(-50%)",
            }
          }
          tabIndex={null}
        />
      </Handle>
      <Marks
        className="rc-slider-mark"
        included={false}
        lowerBound={0}
        marks={Object {}}
        max={11}
        min={0}
        onClickLabel={[Function]}
        reverse={false}
        upperBound={0}
        vertical={false}
      >
        <div
          className="rc-slider-mark"
        />
      </Marks>
    </div>
  </ComponentEnhancer(undefined)>
</HourSlider>
`;

exports[`should handle valid forecast response 1`] = `
<HourSlider
  forecastURL=""
  latitude={51.505455}
  longitude={-0.075356}
  onForecastError={[MockFunction]}
  onForecastReceived={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "daily": Array [
              Object {
                "icon": "rain",
                "summary": "Light rain",
                "temperatureMax": 13.47,
                "temperatureMin": 6.37,
                "time": 1582416000,
              },
              Object {
                "icon": "cloudy",
                "summary": "Overcast",
                "temperatureMax": 12.42,
                "temperatureMin": 4.23,
                "time": 1582516000,
              },
            ],
            "hourly": Array [
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.85,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582848000,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582851600,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582855200,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582858800,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582862400,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582866000,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582869600,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582873200,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582876800,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582880400,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582884000,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582887600,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582891200,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582894800,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582898400,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582902000,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582905600,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582909200,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582912800,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582916400,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582920000,
                "windSpeed": 6.31,
              },
              Object {
                "humidity": 0.82,
                "icon": "rain",
                "precipProbability": 0.82,
                "summary": "Rain",
                "temperature": 7.34,
                "time": 1582923600,
                "windSpeed": 6.31,
              },
            ],
            "timezone": "Europe/London",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onSliderChange={[MockFunction]}
>
  <ComponentEnhancer(undefined)
    activeDotStyle={Object {}}
    className=""
    disabled={true}
    dotStyle={Object {}}
    dots={false}
    handle={[Function]}
    handleStyle={
      Array [
        Object {},
      ]
    }
    included={false}
    marks={Object {}}
    max={11}
    min={0}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    railStyle={Object {}}
    reverse={false}
    step={1}
    trackStyle={
      Array [
        Object {},
      ]
    }
    vertical={false}
  >
    <div
      className="rc-slider rc-slider-disabled"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <Track
        className="rc-slider-track"
        included={false}
        length={0}
        offset={0}
        reverse={false}
        style={Object {}}
        vertical={false}
      />
      <Steps
        activeDotStyle={Object {}}
        dotStyle={Object {}}
        dots={false}
        included={false}
        lowerBound={0}
        marks={Object {}}
        max={11}
        min={0}
        prefixCls="rc-slider"
        reverse={false}
        step={1}
        upperBound={0}
        vertical={false}
      >
        <div
          className="rc-slider-step"
        />
      </Steps>
      <Handle
        className="rc-slider-handle"
        disabled={true}
        key="0"
        max={11}
        min={0}
        offset={0}
        prefixCls="rc-slider"
        reverse={false}
        style={Object {}}
        value={0}
        vertical={false}
      >
        <div
          aria-disabled={true}
          aria-valuemax={11}
          aria-valuemin={0}
          aria-valuenow={0}
          className="rc-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="slider"
          style={
            Object {
              "left": "0%",
              "right": "auto",
              "transform": "translateX(-50%)",
            }
          }
          tabIndex={null}
        />
      </Handle>
      <Marks
        className="rc-slider-mark"
        included={false}
        lowerBound={0}
        marks={Object {}}
        max={11}
        min={0}
        onClickLabel={[Function]}
        reverse={false}
        upperBound={0}
        vertical={false}
      >
        <div
          className="rc-slider-mark"
        />
      </Marks>
    </div>
  </ComponentEnhancer(undefined)>
</HourSlider>
`;

exports[`should handle when response contains less hourly data than expected 1`] = `
<HourSlider
  forecastURL=""
  latitude={51.505455}
  longitude={-0.075356}
  onForecastError={
    [MockFunction] {
      "calls": Array [
        Array [
          "Hourly forecast not covered",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onForecastReceived={[MockFunction]}
  onSliderChange={[MockFunction]}
>
  <ComponentEnhancer(undefined)
    activeDotStyle={Object {}}
    className=""
    disabled={true}
    dotStyle={Object {}}
    dots={false}
    handle={[Function]}
    handleStyle={
      Array [
        Object {},
      ]
    }
    included={false}
    marks={Object {}}
    max={11}
    min={0}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    railStyle={Object {}}
    reverse={false}
    step={1}
    trackStyle={
      Array [
        Object {},
      ]
    }
    vertical={false}
  >
    <div
      className="rc-slider rc-slider-disabled"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <Track
        className="rc-slider-track"
        included={false}
        length={0}
        offset={0}
        reverse={false}
        style={Object {}}
        vertical={false}
      />
      <Steps
        activeDotStyle={Object {}}
        dotStyle={Object {}}
        dots={false}
        included={false}
        lowerBound={0}
        marks={Object {}}
        max={11}
        min={0}
        prefixCls="rc-slider"
        reverse={false}
        step={1}
        upperBound={0}
        vertical={false}
      >
        <div
          className="rc-slider-step"
        />
      </Steps>
      <Handle
        className="rc-slider-handle"
        disabled={true}
        key="0"
        max={11}
        min={0}
        offset={0}
        prefixCls="rc-slider"
        reverse={false}
        style={Object {}}
        value={0}
        vertical={false}
      >
        <div
          aria-disabled={true}
          aria-valuemax={11}
          aria-valuemin={0}
          aria-valuenow={0}
          className="rc-slider-handle"
          onBlur={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="slider"
          style={
            Object {
              "left": "0%",
              "right": "auto",
              "transform": "translateX(-50%)",
            }
          }
          tabIndex={null}
        />
      </Handle>
      <Marks
        className="rc-slider-mark"
        included={false}
        lowerBound={0}
        marks={Object {}}
        max={11}
        min={0}
        onClickLabel={[Function]}
        reverse={false}
        upperBound={0}
        vertical={false}
      >
        <div
          className="rc-slider-mark"
        />
      </Marks>
    </div>
  </ComponentEnhancer(undefined)>
</HourSlider>
`;
